### ISSUE TITLE
'int' object has no attribute 'year'

### ISSUE THREAD

**Issue by jaraco** [jaraco/calendra#11]
On Python 3.8, many of the tests fail with the error `'int' object has no attribute 'year'` when adding a Holiday object to a timedelta.

**Comment by jaraco** [comment]
The issue can be distilled to this:

```
>>> from calendra import core
>>> import datetime
>>> core.Holiday(datetime.datetime(2019,12,20)) + datetime.timedelta(minutes=3)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/jaraco/code/main/calendra/calendra/core.py", line 73, in __new__
    cls, date.year, date.month, date.day)
AttributeError: 'int' object has no attribute 'year'
``

**Comment by jaraco** [comment]
The issue seems to have originated in [bpo-32417](https://bugs.python.org/issue32417).

**Comment by pganssle** [comment]
Interesting. I think the simplest thing to do is to define a `__add__` on `core.Holiday` that returns the base datetime class, if you want the old behavior.

**Comment by jaraco** [comment]
I started down that route, but it wasn't clear to me what that implementation would look like... especially when the `other` parameter to `__add__` might not be a timedelta, but might be a relativedelta or something else, so I've decided instead to try to conform the expected signature.

**Comment by jaraco** [comment]
One thing I did observe - the Holiday class already supports use of `.replace()` to create new instances with different values (but retaining other properties). I wonder if the "add timedelta" functionality had instead used `orig.replace(...)`, if that would be a more robust way to construct a new instance from an old one (and give subclasses more control over how a new instance of one's own type could be constructed).

**Comment by pganssle** [comment]
@jaraco Won't that break all your existing callers, though?

I think the easiest thing to do if you want to continue supporting the old interface is to write your `__add__` like so:

```python
    def _asdatetime(self):
        return datetime(self.year, self.month, self.day,
                        self.hour, self.minute, self.second, self.microsecond,
                        fold=self.fold, tzinfo=self.tzinfo)

    def __add__(self, other):
        return self._asdatetime().__add__(other)
```

Depending on how much you care about efficiency, there may be faster ways to get the equivalent datetime object.

If you don't really care about supporting the old interface, then your solution seems to work fine too.

**Comment by jaraco** [comment]
Yes, good point about compatibility. I did add support for compatibility in 97214a01cc8ab31dd7edd97c091e87aae8b569d6.

**Comment by jaraco** [comment]
Inspired by your suggestion, I implemented an alternate fix in 574746e, which needed `__add__` and `__sub__`, but otherwise was minimally invasive.
